tests/extensions/test_grid.py

```diff
@@ -86,7 +86,7 @@ def test_clear_code(self) -> None:
         GridExtension.ext(self.item).apply(code)
 
         with self.assertRaises(ValueError):
-            GridExtension.ext(self.item).code = None
+            GridExtension.ext(self.item).code = ""
```

Comment:
either attempting to set this to None or empty string should raise a ValueError, so the `None` case should remain instead of being replaced. Also some tests for non-compliant code strings should be added. There is a regex defined in GridExtension as to what they need to match.

Reply:
Should we update the setter signature to take an `Optional[str]` so we don't make type checkers angry? No change to functionality (still raises `ValueError`).

> Also some tests for non-compliant code strings should be added.

Do you have a couple off the top of your head? Just stuff like `this-is-not-a-grid-code`?

Reply:
or accidentally put the epsg number in there.